virtual_ecosystem/models/animal/animal_model.py

```diff
@@ -148,45 +212,37 @@ def _setup_grid_neighbors(self) -> None:
         """
         self.data.grid.set_neighbours(distance=sqrt(self.data.grid.cell_area))
 
-    def get_community_by_key(self, key: int) -> AnimalCommunity:
-        """Function to return the AnimalCommunity present in a given grid square.
-
-        This function exists principally to provide a callable for AnimalCommunity.
-
-        Args:
-            key: The specific grid square integer key associated with the community.
-
-        Returns:
-            The AnimalCommunity object in that grid square.
-
-        """
-        return self.communities[key]
-
     def _initialize_communities(self, functional_groups: list[FunctionalGroup]) -> None:
-        """Initialize the animal communities.
+        """Initialize the animal communities by creating and populating animal cohorts.
 
         Args:
             functional_groups: The list of functional groups that will populate the
             model.
         """
+        # Initialize communities dictionary with cell IDs as keys and empty lists for
+        # cohorts
+        self.communities = {cell_id: list() for cell_id in self.data.grid.cell_id}
+
+        # Iterate over each cell and functional group to create and populate cohorts
+        for cell_id in self.data.grid.cell_id:
+            for functional_group in functional_groups:
+                # Calculate the number of individuals using Damuth's Law
+                individuals = damuths_law(
+                    functional_group.adult_mass, functional_group.damuths_law_terms
+                )
 
-        # Generate a dictionary of AnimalCommunity objects, one per grid cell.
-        self.communities = {
-            k: AnimalCommunity(
-                functional_groups=functional_groups,
-                data=self.data,
-                community_key=k,
-                neighbouring_keys=list(self.data.grid.neighbours[k]),
-                get_destination=self.get_community_by_key,
-                constants=self.model_constants,
-            )
-            for k in self.data.grid.cell_id
-        }
-
-        # Create animal cohorts in each grid square's animal community according to the
-        # populate_community method.
-        for community in self.communities.values():
-            community.populate_community()
+                # Create a cohort of the functional group
+                cohort = AnimalCohort(
+                    functional_group=functional_group,
+                    mass=functional_group.adult_mass,
+                    age=0.0,
+                    individuals=individuals,
+                    centroid_key=cell_id,
+                    grid=self.data.grid,
+                    constants=self.model_constants,
+                )
```

Comment:
As far as I can tell, all cohorts in all grid cells are identical. Where do you set - if you do - the initial distribution of individuals across the simulation domain?

Reply:
There is not currently a setting for number of individuals across the entire grid. The number of individuals are determined through population densities and the spatial range of the simulation. I expect this format to be much more commonly used but we may need a "fixed number of individuals" input later, particularly for species of special interest and reintroduction programs.